templates/openpanel/index.ts

```diff
@@ -0,0 +1,205 @@
+import {
+  Output,
+  randomPassword,
+  randomString,
+  Services,
+} from "~templates-utils";
+import { Input } from "./meta";
+
+export function generate(input: Input): Output {
+  const services: Services = [];
+  const postgresPassword = randomPassword();
+  const cookieSecret = randomString(32);
+  const redisPassword = randomPassword();
+  const basicAuthHash = randomString(8);
+
+  const commonEnvVars = [
+    `NODE_ENV=production`,
+    `SELF_HOSTED=true`,
+    `GEO_IP_HOST=http://$(PROJECT_NAME)_${input.appServiceName}-geo:8080`,
+    `BATCH_SIZE=5000`,
+    `BATCH_INTERVAL=10000`,
+    `ALLOW_REGISTRATION=true`,
+    `ALLOW_INVITATION=true`,
+    `REDIS_URL=redis://default:${redisPassword}@$(PROJECT_NAME)-${input.appServiceName}-redis:6379`,
+    `CLICKHOUSE_URL=http://$(PROJECT_NAME)_${input.appServiceName}-ch:8123`,
+    `DATABASE_URL=postgresql://postgres:${postgresPassword}@$(PROJECT_NAME)_${input.appServiceName}-db:5432/$(PROJECT_NAME)?schema=public`,
+    `DATABASE_URL_DIRECT=postgresql://postgres:${postgresPassword}@$(PROJECT_NAME)_${input.appServiceName}-db:5432/$(PROJECT_NAME)?schema=public`,
+    `NEXT_PUBLIC_DASHBOARD_URL=https://${input.appDomain}`,
+    `NEXT_PUBLIC_API_URL=https://${input.appDomain}/api`,
+    `COOKIE_SECRET=${cookieSecret}`,
+    `EMAIL_SENDER=${input.emailSender || "noreply@example.com"}`,
+    input.resendApiKey
+      ? `RESEND_API_KEY=${input.resendApiKey}`
+      : `# RESEND_API_KEY=`,
+  ];
+
+  const caddyFile = `
+http://{$PRIMARY_DOMAIN} {
+    
+    handle_path /api* {
+        reverse_proxy {$API_URL}:3000
+    }
+
+    reverse_proxy /* {$DASHBOARD_URL}:3000
+}
+
+http://worker.{$PRIMARY_DOMAIN} {
+
+    basic_auth {
+        admin ${basicAuthHash}
```

Comment:
a random hash will not work, basic auth hash is based on the password you want to set

also, we have this built in our services, so no need to add it in caddy

also, we don't need caddy, just use domains for each service